packages/frontend-v2/hooks/usePlan.ts

```diff
@@ -0,0 +1,35 @@
+import useSWR, { SWRResponse } from "swr";
+import { API } from "@graduate/api-client";
+import { GetPlanResponse } from "@graduate/common";
+import axios, { AxiosError } from "axios";
+import { useRedirectUnAuth } from "./utils";
+
+type planResponse = SWRResponse<GetPlanResponse, AxiosError>;
+
+interface UsePlanReturn {
+  plan?: planResponse["data"];
+  error?: planResponse["error"];
+  isLoading: boolean;
+  mutatePlan: planResponse["mutate"];
+}
+
+export function usePlan(planId: number, jwt: string): UsePlanReturn {
+  const key = `api/plans/${planId}`;
+  const redirect = useRedirectUnAuth();
+  const {
+    data: plan,
+    error,
+    mutate: mutatePlan,
+  } = useSWR(key, async () => await API.plans.get(planId, jwt));
+
+  if (axios.isAxiosError(error)) {
+    redirect(error);
+  }
+
+  return {
+    plan,
+    error,
+    mutatePlan,
+    isLoading: !plan && !error,
+  };
```

Comment:
a trick i like doing is something like this, though idk how the types would work out
```typescript
const { data, mutate, ...rest } = useSwr(...);
return { ...rest, plan: data, mutatePlan: mutate };
```

Reply:
I agree with this, for the type you can check what the return type for useSWR is. That may be easier than having the interface since UsePlanReturn can just be the same type returned by `useSWR<return type of fetcher, error type thrown by fetcher>`. Not quite sure of the generics, you'll have to take a deeper look into it.